File: lokki/src/main/scala/omigo_core/FilePathsReader.scala
package omigo_core

class FilePathsReader(val filePaths: List[String]) {
  var curIndex = 0

  def hasNext() = {
    // TODO: changing the logic. Need to reflect in python
    curIndex != -1 && curIndex < filePaths.length
  }

  def close(): Unit = {
    // TODO: python code has issue
  }

  def next(): String = {
    if (hasNext() == false)
      return null 

    // advance the pointer for next time
    val result = filePaths(curIndex)
    curIndex = curIndex + 1

    // check for end
    if (curIndex >= filePaths.length) {
      // TODO : no need to do this
      // filepaths = None
      curIndex = -1
    }

    // print which file is processed
    println("FilePathsReader: processing file: " + result)
    result
  } 
}

File: python-packages/hydra/src/omigo_hydra/file_paths_reader.py
"""FilePathsReader class"""

from omigo_core import utils

class FilePathsReader:
    """A simple class to read filepaths data"""

    # constructor
    def __init__(self, filepaths):
        # filepaths is the array of paths
        self.filepaths = filepaths
        # cur_index is the pointer to the current index in filepaths
        self.cur_index = 0

    # has next returns true if there is still data to be read
    def has_next(self):
        return self.filepaths is not None and self.cur_index < len(self.filepaths)

    # close the file reader
    def close(self):
        self.cur_index = len(self.filepaths) + 1

    # next returns the next data. None otherwise
    def next(self):
        if (self.has_next() == False):
            return None

        # advance the pointer for next time
        result = self.filepaths[self.cur_index]
        self.cur_index = self.cur_index + 1

        # check for end
        if (self.cur_index >= len(self.filepaths)):
            self.filepaths = None
            self.cur_index = -1

        # print which file is processed
        utils.info("FilePathsReader: processing file: {}".format(result))
        return result


